### ISSUE TITLE
"getsockopt: connection refused" in protokube logs on nodes

### ISSUE THREAD

**Issue by juris** [kubernetes/kops#4590]
The following error constantly shows up in protokube logs on worker nodes (everything is fine on masters):
```
I0306 16:19:43.794606    1178 channels.go:45] Running command: channels apply channel s3://k8s-state/dummy.k8s.local/addons/bootstrap-channel.yaml --v=4 --yes
I0306 16:19:43.861374    1178 channels.go:48] error running channels apply channel s3://k8s-state/dummy.k8s.local/addons/bootstrap-channel.yaml --v=4 --yes:
I0306 16:19:43.861408    1178 channels.go:49] Error: error querying kubernetes version: Get http://localhost:8080/version: dial tcp [::1]:8080: getsockopt: connection refused
```

---
1. What `kops` version are you running? The command `kops version`, will display
 this information.

kops 1.8.1

2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.

```
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.3", GitCommit:"d2835416544f298c919e2ead3be3d0864b52323b", GitTreeState:"clean", BuildDate:"2018-02-09T21:51:06Z", GoVersion:"go1.9.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"8", GitVersion:"v1.8.8", GitCommit:"2f73858c9e6ede659d6828fe5a1862a48034a0fd", GitTreeState:"clean", BuildDate:"2018-02-09T21:23:25Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
```

3. What cloud provider are you using?

AWS

4. What commands did you run?  What is the simplest way to reproduce this issue?

ssh to any node and run `docker logs <protokube_container_id>`

5. What happened after the commands executed?

Errors in container log.

6. What did you expect to happen?

There should not be any errors in container log.

7. Please provide your cluster manifest. Execute
  `kops get --name my.example.com -oyaml` to display your cluster manifest.
  You may want to remove your cluster name and other sensitive information.

```
apiVersion: kops/v1alpha2
kind: Cluster
metadata:
  creationTimestamp: 2018-01-22T11:15:26Z
  name: dummy.k8s.local
spec:
  api:
    loadBalancer:
      type: Internal
  authorization:
    alwaysAllow: {}
  channel: stable
  cloudProvider: aws
  configBase: s3://k8s-state/dummy.k8s.local
  etcdClusters:
  - etcdMembers:
    - instanceGroup: master-us-east-1a
      name: a
    - instanceGroup: master-us-east-1b
      name: b
    - instanceGroup: master-us-east-1c
      name: c
    name: main
  - etcdMembers:
    - instanceGroup: master-us-east-1a
      name: a
    - instanceGroup: master-us-east-1b
      name: b
    - instanceGroup: master-us-east-1c
      name: c
    name: events
  iam:
    allowContainerRegistry: true
    legacy: false
  kubernetesApiAccess:
  - 10.3.0.0/16
  kubernetesVersion: 1.8.8
  masterInternalName: api.internal.dummy.k8s.local
  masterPublicName: api.dummy.k8s.local
  networkCIDR: 10.3.0.0/16
  networking:
    calico:
      crossSubnet: true
  nonMasqueradeCIDR: 100.64.0.0/10
  sshAccess:
  - 10.3.0.0/16
  subnets:
  - cidr: 10.3.0.0/24
    name: masters-us-east-1a
    type: Private
    zone: us-east-1a
  - cidr: 10.3.1.0/24
    name: masters-us-east-1b
    type: Private
    zone: us-east-1b
  - cidr: 10.3.2.0/24
    name: masters-us-east-1c
    type: Private
    zone: us-east-1c
  - cidr: 10.3.3.0/24
    name: utility-us-east-1a
    type: Utility
    zone: us-east-1a
  - cidr: 10.3.4.0/24
    name: utility-us-east-1b
    type: Utility
    zone: us-east-1b
  - cidr: 10.3.5.0/24
    name: utility-us-east-1c
    type: Utility
    zone: us-east-1c
  - cidr: 10.3.8.0/21
    name: nodes-us-east-1a
    type: Private
    zone: us-east-1a
  topology:
    dns:
      type: Private
    masters: private
    nodes: private

---

apiVersion: kops/v1alpha2
kind: InstanceGroup
metadata:
  creationTimestamp: 2018-01-22T11:15:26Z
  labels:
    kops.k8s.io/cluster: dummy.k8s.local
  name: master-us-east-1a
spec:
  image: ami-ce15e9b3
  machineType: t2.medium
  maxSize: 1
  minSize: 1
  nodeLabels:
    kops.k8s.io/instancegroup: master-us-east-1a
  role: Master
  subnets:
  - masters-us-east-1a

---

apiVersion: kops/v1alpha2
kind: InstanceGroup
metadata:
  creationTimestamp: 2018-01-22T11:15:27Z
  labels:
    kops.k8s.io/cluster: dummy.k8s.local
  name: master-us-east-1b
spec:
  image: ami-ce15e9b3
  machineType: t2.medium
  maxSize: 1
  minSize: 1
  nodeLabels:
    kops.k8s.io/instancegroup: master-us-east-1b
  role: Master
  subnets:
  - masters-us-east-1b

---

apiVersion: kops/v1alpha2
kind: InstanceGroup
metadata:
  creationTimestamp: 2018-01-22T11:15:27Z
  labels:
    kops.k8s.io/cluster: dummy.k8s.local
  name: master-us-east-1c
spec:
  image: ami-ce15e9b3
  machineType: t2.medium
  maxSize: 1
  minSize: 1
  nodeLabels:
    kops.k8s.io/instancegroup: master-us-east-1c
  role: Master
  subnets:
  - masters-us-east-1c

---

apiVersion: kops/v1alpha2
kind: InstanceGroup
metadata:
  creationTimestamp: 2018-01-22T11:15:27Z
  labels:
    kops.k8s.io/cluster: dummy.k8s.local
  name: nodes
spec:
  image: ami-ce15e9b3
  machineType: m4.large
  maxSize: 2
  minSize: 2
  nodeLabels:
    kops.k8s.io/instancegroup: nodes
  role: Node
  subnets:
  - nodes-us-east-1a

---

apiVersion: kops/v1alpha2
kind: InstanceGroup
metadata:
  creationTimestamp: 2018-01-22T11:37:39Z
  labels:
    kops.k8s.io/cluster: dummy.k8s.local
  name: nodes-backend-1a
spec:
  image: ami-ce15e9b3
  machineType: m4.xlarge
  maxSize: 4
  minSize: 4
  nodeLabels:
    kops.k8s.io/instancegroup: nodes-backend-1a
  role: Node
  subnets:
  - nodes-us-east-1a
```

9. Anything else do we need to know?

Hosts are running with CoreOS 1632.3.0

**Comment by chrislovecnm** [comment]
/assign @KashifSaadat @gambol99 

lets have the CoreOS gurus look at this one

**Comment by chrislovecnm** [comment]
@KashifSaadat / @gambol99 any ideas?

**Comment by gambol99** [comment]
the fact the worker nodes are running protokube is very odd!! as i removed it myself [here](https://github.com/gambol99/kops/blob/master/nodeup/pkg/model/protokube.go#L51-L55) in [#PR3091](https://github.com/kubernetes/kops/pull/3091) (i.e. non-master only if your using gossip dns).  As for the error this is valid, it's caused [here](https://github.com/gambol99/kops/blob/master/channels/pkg/cmd/apply_channel.go#L60) when protokube spins off a `channels` to apply the addons; attempts to speak to the API on insecure port and since this isn't a master fails. @juris .. was this an update or a new cluster build?

**Comment by chrislovecnm** [comment]
@gambol99 he is running a gossip cluster, so I think we need protokube running on the workers

**Comment by gambol99** [comment]
arrhh, my bad :-) ...didn't see anything related to gossip in the cluster spec so assumed route53, just saw it's based on [dns](https://github.com/kubernetes/kops/blob/master/pkg/dns/gossip.go#L22) though. I've raised [#PR4651](https://github.com/kubernetes/kops/pull/4651) which stops running `channels` on a non-master

**Comment by KashifSaadat** [comment]
Will close this, I believe it's resolved as part of PR #4651. @juris feel free to re-open if you're still experiencing this problem with the latest kops release (v1.9.0).

/close
